kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/DefaultSharedIndexInformerTest.java

```diff
@@ -927,6 +928,23 @@ void testGenericKubernetesResourceSharedIndexInformerWithNamespaceConfigured() t
     assertEquals(0, foundExistingAnimal.getCount());
   }
 
+  @Test
+  void testGenericKubernetesResourceSharedIndexInformerWithAdditionalDeserializers() throws InterruptedException {
+    // Given
+    setupMockServerExpectations(Animal.class, "ns1", this::getList, r -> new WatchEvent(getAnimal("red-panda", "Carnivora", r), "ADDED"), null, null);
+
+    // When
+    KubernetesDeserializer.registerCustomKind("jungle.example.com/v1","Animal", CronTab.class);
```

Comment:
This is a little "extreme", instead of registering an incompatible version I'm registering a completely wrong deserializer that will take precedence; but it shows the behaviour described in #3785